openssl/src/bn.rs

```diff
@@ -59,30 +99,78 @@ impl BigNumContext {
     }
 }
 
+foreign_type! {
+    type CType = ffi::BIGNUM;
+    fn drop = ffi::BN_free;
+
+    /// Dynamically sized large number impelementation
+    ///
+    /// Perform large number mathematics.  Create a new BigNum
+    /// with [`new`].  Perform stanard mathematics on large numbers using
+    /// methods from [`Dref<Target = BigNumRef>`]
+    ///
+    /// OpenSSL documenation at [`BN_new`].
+    ///
+    /// [`new`]: struct.BigNum.html#method.new
+    /// [`Dref<Target = BigNumRef>`]: struct.BigNum.html#deref-methods
+    /// [`BN_new`]: https://www.openssl.org/docs/man1.1.0/crypto/BN_new.html
+    ///
+    /// # Examples
+    /// ```
+    /// use openssl::bn::BigNum;
+    /// # use openssl::error::ErrorStack;
+    /// # fn bignums() -> Result< (), ErrorStack > {
+    /// let little_big = BigNum::from_u32(std::u32::MAX)?;
+    /// assert_eq!(*&little_big.num_bytes(), 4);
+    /// # Ok(())
+    /// # }
+    /// # fn main () { bignums(); }
+    /// ```
+    pub struct BigNum;
+    /// Reference to a [`BigNum`]
+    ///
+    /// [`BigNum`]: struct.BigNum.html
+    pub struct BigNumRef;
+}
+
 impl BigNumRef {
     /// Erases the memory used by this `BigNum`, resetting its value to 0.
     ///
     /// This can be used to destroy sensitive data such as keys when they are no longer needed.
+    ///
+    /// OpenSSL documentation at [`BN_clear`]
+    ///
+    /// [`BN_clear`]: https://www.openssl.org/docs/man1.1.0/crypto/BN_clear.html
     pub fn clear(&mut self) {
         unsafe { ffi::BN_clear(self.as_ptr()) }
     }
 
-    /// Adds a `u32` to `self`.
+    /// Adds a `u32` to `self`.  OpenSSL documentation at [`BN_add_word`]
```

Comment:
Can you put the documentation link in a separate paragraph for all of the methods here?